docs/articles_en/openvino-workflow/running-inference/inference-devices-and-modes/gpu-device/remote-tensor-api-gpu-plugin.rst

```diff
@@ -350,7 +350,139 @@ For more details, see the code snippets below:
 
 The ``ov::intel_gpu::ocl::D3DContext`` and ``ov::intel_gpu::ocl::VAContext`` classes are derived from ``ov::intel_gpu::ocl::ClContext``.
 Therefore, they provide the functionality described above and extend it
-to allow creation of ``ov::RemoteTensor`` objects from ``ID3D11Buffer``, ``ID3D11Texture2D`` pointers or the ``VASurfaceID`` handle respectively.
+to allow creation of ``ov::RemoteTensor`` objects from ``ID3D11Buffer``, ``ID3D11Texture2D``
+pointers or the ``VASurfaceID`` handle respectively, as shown in the examples below:
+
+
+.. tab-set::
+
+   .. tab-item:: ID3D11Buffer
+      :sync: id3d11-buffer
+
+      .. code-block:: cpp
+
+         // ...
+
+         // initialize the core and load the network
+         ov::Core core;
+         auto model = core.read_model("model.xml");
+         auto compiled_model = core.compile_model(model, "GPU");
+         auto infer_request = compiled_model.create_infer_request();
+
+
+         // obtain the RemoteContext from the compiled model object and cast it to D3DContext
+         auto gpu_context = compiled_model.get_context().as<ov::intel_gpu::ocl::D3DContext>();
+         // obtain the D3D context handle from the RemoteContext,
+         // get device info and create a queue
+         cl::Context cl_context = gpu_context;
+         cl::Device device = cl::Device(cl_context.getInfo<CL_CONTEXT_DEVICES>()[0].get(), true);
+         cl_command_queue_properties props = CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE;
+         cl::CommandQueue queue = cl::CommandQueue(cl_context, device, props);
+
+         // create the D3D buffer within the obtained context
+         auto input = model->get_parameters().at(0);
+         auto input_size = ov::shape_size(input->get_shape());
+         cl_int err;
+         cl::Buffer shared_buffer(cl_context, CL_MEM_READ_WRITE, input_size, NULL, &err);
+         // wrap the buffer into RemoteBlob
+         auto shared_blob = gpu_context.create_tensor(input->get_element_type(), input->get_shape(), shared_buffer);
+
+         // ...
+         // execute user kernel
+         cl::Program program;
+         cl::Kernel kernel(program, "user_kernel");
+         kernel.setArg(0, shared_buffer);
+         queue.enqueueNDRangeKernel(kernel,
+                                    cl::NDRange(0),
+                                    cl::NDRange(input_size),
+                                    cl::NDRange(1),
+                                    nullptr,
+                                    nullptr);
+         queue.finish();
```

Comment:
I think this is not relevant code for D3D buffer sharing. I believe we need something like this:

```
auto input = model->get_parameters().at(0);
ID3D11Buffer* d3d_handle = get_d3d_buffer();
auto tensor = gpu_context.create_tensor(input->get_element_type(), input->get_shape(), d3d_handle);
infer_request.set_tensor(input, tensor);

```